Gems/Atom/Feature/Common/Code/Source/Mesh/ModelReloader.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Source/Mesh/ModelReloader.h>
+#include <Atom/RPI.Reflect/Model/ModelAsset.h>
+#include <Atom/RPI.Public/Model/Model.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        ModelReloader::ModelReloader(
+            Data::Asset<RPI::ModelAsset> modelAsset, RemoveModelFromReloaderSystemEvent::Handler& removeReloaderFromSystemHandler)
+        {
+            m_modelAsset.push_back(modelAsset);
+            m_pendingDependencyListStatus.reset();
+            removeReloaderFromSystemHandler.Connect(m_onRemoveReloaderFromSystem);
+
+            // Iterate over the model and track the assets that need to be reloaded
+            for (auto& modelLodAsset : modelAsset->GetLodAssets())
+            {
+                for (auto& mesh : modelLodAsset->GetMeshes())
+                {
+                    for (auto& streamBufferInfo : mesh.GetStreamBufferInfoList())
+                    {
+                        InsertMeshDependencyIfUnique(streamBufferInfo.m_bufferAssetView.GetBufferAsset());
+                        
+                    }
+                    InsertMeshDependencyIfUnique(mesh.GetIndexBufferAssetView().GetBufferAsset());
+                }
+                m_modelDependencies.push_back(modelLodAsset);
+            }
+
+            AZ_Assert(
+                m_meshDependencies.size() <= m_pendingDependencyListStatus.size(),
+                "There are more buffers used by the model %s than are supported by the ModelReloader.", modelAsset.GetHint().c_str());
```

Comment:
Same question, is this programmer's fault or data driven error, if the latter then this should be a AZ_Error. If it's data driven also the editor is suppose to handle this, not crash, maybe this reloader should be put into an invalid state and the system should remove it.

Reply:
This would be programmer's fault for using a value in the bitset size that doesn't exceed the maximum supported number of vertex streams.